### PR TITLE
Android-Support to AndroidX Upgrade

### PR DESCRIPTION
**Pull request by PWRxPSYCHO**
After looking into upgrading NGA-Geopackage, it seems that there are too many clashing dependencies from android.support and androidx. I saw there was a feature request to update the older libraries to the newer libraries. 

PR: #1853 - should be done after this to resolve the duplicate file issues

Issue: #1453 

I followed this guide: https://developer.android.com/jetpack/androidx/migrate/artifact-mappings to update all legacy mappings to the equivalent AndroidX

### CODE REVIEW

**Comment by PWRxPSYCHO** [comment]
https://github.com/osmdroid/osmdroid/actions/runs/3245835164/jobs/5328503169#step:5:1338

I will investigate this unit test error.

**Comment by PWRxPSYCHO** [comment]
> https://github.com/osmdroid/osmdroid/actions/runs/3245835164/jobs/5328503169#step:5:1338
> 
> I will investigate this unit test error.


It looks like all the tests are passing. There is one test skipped in the WMS parser but that module is still in beta so I don't see an issue with it skipping.
 
Screenshot 2022-10-14 at 08-58-47 Test results - Test Summary
Screenshot 2022-10-14 at 08-58-34 Test results - Test Summary
Screenshot 2022-10-14 at 08-58-16 Test results - Test Summary

File: gradle.properties [hunk 1]
```diff
@@ -41,12 +41,12 @@ gradleFury.version=1.0.14
 
 
 android-plugin.version=7.1.3
-android-support.version=28.0.0
 # To be used for packages that depend on support lib (minSDK = 14 since 26.0.0)
 android-minSdkForSupportLib.version=14
 junit.version=4.13.2
 robolectric.version=4.7.3
 android.useAndroidX=true
+android.enableJetifier=true
```
Comment (monsieurtanuki) [inline gradle.properties:49]: Just a minor remark from https://developer.android.com/jetpack/androidx/migrate:
> Additionally, the [enableJetifier](https://developer.android.com/jetpack/androidx/migrate#migrate_an_existing_project_using_android_studio) flag mentioned in this guide can lead to slower build times and should not be used unless it’s necessary.

**Review by monsieurtanuki — APPROVED** [review]
Hi @PWRxPSYCHO!
From what I read, this PR looks good to me.
The thing is that I have no idea what the impact of that migration would be on legacy apps.
I assume that in 2022 that should be OK, as long as the min sdk is not changed.

File: gradle.properties (hunk 1 above)
Comment (PWRxPSYCHO) [reply gradle.properties:49]: I will address this!

Comment (PWRxPSYCHO) [reply gradle.properties:49]: Legacy Library

This seems to be the only thing that requires the enableJetifier flag on. The logging would need to be updated as newer versions of ACRA removed `setReportSender()` which is used in `OsmApplication.java`

Comment (PWRxPSYCHO) [reply gradle.properties:49]: Overall the build time does not seem to be severely impacted by this:
As indicated here
Build

Total time

Comment (monsieurtanuki) [reply gradle.properties:49]: Thank you @PWRxPSYCHO for your detailed report!

If I sum up:
* the impact of `android.enableJetifier=true` looks minimal, and is _de facto_ limited to the sample app
* we may create an issue about getting rid of `android.enableJetifier=true` that would involve an upgrade of `ch.acra` and some minor refactoring around it

**Comment by PWRxPSYCHO** [comment]
> Hi @PWRxPSYCHO! From what I read, this PR looks good to me. The thing is that I have no idea what the impact of that migration would be on legacy apps. I assume that in 2022 that should be OK, as long as the min sdk is not changed.

@monsieurtanuki I did my best to maintain legacy mappings so they should be 1-1 and cause no breaking changes as far as legacy apps are concerned. The min sdk has not been touched. I did go through all the versions of the dependencies to make sure everything was still compatible.

**Comment by eclectice** [comment]
What is the status of this pull request, and when will it be merged into the master branch? I'd like to write a simple app in Kotlin that makes use of **ViewPager2**.

Only the **osmdroid:master** branch has been tested with the Android Gradle plugin (AGP) 7.5.1, which requires a Java 11 JVM, and it works flawlessly. The broken Simple Map app in this OsmDroid project continues to puzzle me; I remain of the opinion that having the OpenStreetMapViewer app built is adequate.

On the other hand, did you encounter any ***prepareKotlinBuildScriptModel*** error when building **OpenStreetMapViewer** taken from **osmdroid:master** branch with Android Studio (Android Studio Dolphin | 2021.3.1 | Build #AI-213.7172.25.2113.9014738)? 

```
Task 'prepareKotlinBuildScriptModel' not found in project ':OpenStreetMapViewer'
```

To remove the error, I need to include this Gradle groovy snippet in OpenStreetMapViewer's build.gradle:

```groovy
tasks.register("prepareKotlinBuildScriptModel"){}
```

**Comment by monsieurtanuki** [comment]
@spyhunter99 It's probably time to grant merge or/and admin privileges (whatever they're called in github) to contributors that have skills, time and motivation, like @eclectice.

**Comment by spyhunter99** [comment]
invite sent

**Comment by eclectice** [comment]
> @spyhunter99 It's probably time to grant merge or/and admin privileges (whatever they're called in github) to contributors that have skills, time and motivation, like @eclectice.

> invite sent

Thank you for the invitation, but I will KIV it for the time being because I am still unfamiliar with git and the GitHub versioning system; I am more familiar with the Subversion system.

@spyhunter99 I pulled your merge, but when I tested it on a physical Android 11 device, it had a lot of issues (specifically Samsung Galaxy Tab S7 FE SM-T733). Even for simple things like accessing shared preferences, getting a system service context, and so on, the OpenStreetMapViewer app receives a lot of StrictMode disk read/write violations. To reduce them, I use ***StrictMode.allowTgreadDiskWrites()*** whenever possible. 

```java
        StrictMode.ThreadPolicy oldPolicy = null;
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD) {
            oldPolicy = StrictMode.allowThreadDiskWrites();
        }
        try {
            mSensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
        } finally {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD) {
                StrictMode.setThreadPolicy(oldPolicy);
            }
        }
```

I'm also implementing various permission checks for code that requires runtime permissions. Because there are cleartext network traffic errors when accessing some OpenStreetMap links, I'm adding them to the app's network configuration file. 

My work is still in progress. I will keep you informed when it is done.

**Comment by eclectice** [comment]
When running any sample demo that uses the **GpsMyLocationProvider** on an Android 11 device, it appears that the sample demo receives a leak notification from Leak Canary. Following further investigation, it appears that the Location Service obtains its context from the activity rather than the application context. This is well explained in this article: https://dev.to/pyricau/beware-packagemanager-leaks-223g.

As a result, I changed the majority of the system services in the sample demo to use the application context rather than the activity context, with the exception of ***Context.LAYOUT_INFLATER_SERVICE***, which requires the activity context.

**Pull request merged**